app/livechat/client/views/app/tabbar/customerChatHistory.js

```diff
@@ -0,0 +1,155 @@
+import { Template } from 'meteor/templating';
+import moment from 'moment';
+import { ReactiveVar } from 'meteor/reactive-var';
+import './customerChatHistory.html';
+import './chatRoomHistoryItem.html';
+import './chatRoomSearchItem.html';
+import _ from 'underscore';
+
+import { APIClient } from '../../../../../utils/client';
+
+const ITEMS_COUNT = 50;
+Template.customerChatHistory.helpers({
+
+	hasChatHistory() {
+		// will return if user has any chatHistory or not
+		return Template.instance().hasHistory.get();
+	},
+	isSearching() {
+		return Template.instance().isSearching.get();
+	},
+	isAllChat() {
+		// will return is have to load all chat
+		return Template.instance().isAllChat.get();
+	},
+	isChatClicked() {
+		// will return that if you have clicked in a single chatHistory
+		return Template.instance().isChatClicked.get();
+	},
+	isfound() {
+		// will return if find any search result
+		return Template.instance().isFound.get();
+	},
+	searchResults() {
+		// will return search result
+		return 	Template.instance().searchResult.get();
+	},
+	previousChats() {
+		// will return pervious chats list
+		return Template.instance().history.get();
+	},
+	clickRid() {
+		return Template.instance().clickRid.get();
+	},
+	clickToken() {
+		return Template.instance().clickToken.get();
+	},
+	title() {
+		let title = moment(this.ts).format('L LTS');
+
+		if (this.label) {
+			title += ` - ${ this.label }`;
+		}
+		return title;
+	},
+});
+
+const DEBOUNCE_TIME_FOR_SEARCH_DEPARTMENTS_IN_MS = 300;
+Template.customerChatHistory.onCreated(function() {
+	const currentData = Template.currentData();
+	this.rid = new ReactiveVar(currentData.rid);
+	this.filter = new ReactiveVar('');
+	this.isKeyUp = new ReactiveVar(false);
+	this.hasHistory = new ReactiveVar(false);
+	this.isFound = new ReactiveVar();
+	this.visitorId = new ReactiveVar();
+	this.history = new ReactiveVar([]);
+	this.offset = new ReactiveVar(0);
+	this.total = new ReactiveVar(0);
+	this.isAllChat = new ReactiveVar(true);
+	this.isSearching = new ReactiveVar(false);
+	this.isChatClicked = new ReactiveVar(true);
+	this.clickRid = new ReactiveVar();
+	this.clickToken = new ReactiveVar();
```

Comment:
You don't need this, please remove it.

Reply:
i need this when to store rid and token of the clicked room to access from other template or else i can use session storage for that

Reply:
You don't need this information here, you can pass it through the Template data.